Support for ItemizedPaymentInformation element in ShipmentConfirmRequest

thijsw opened the pull request:
Hi,

I've added support for the `<ItemizedPaymentInformation>` element in a Shipment Confirm Request per the UPS documentation. I've tried to follow the current coding style of the code base.

stefandoorn commented on src/Shipping.php:
```diff
@@ -305,9 +282,51 @@ private function createConfirmRequest(
-//         TODO: create ItemizedPaymentInformation class and required subclasses for node processing
-//        } elseif ($shipment->getItemizedPaymentInformation()) {
-//            $paymentNode = $shipmentNode->appendChild($xml->createElement('ItemizedPaymentInformation'));
+        } elseif ($shipment->getItemizedPaymentInformation()) {
```
Can we also move this towards using toNode?

thijsw replied:
done

stefandoorn commented on src/Entity/BillShipper.php:
```diff
@@ -18,6 +18,11 @@ class BillShipper
     private $creditCard;
 
     /**
+     * TODO: AlternatePaymentMethod?
```
Is this work in progress?

thijsw replied:
not anymore :)

stefandoorn commented on src/Entity/BillThirdParty.php:
```diff
@@ -63,4 +67,32 @@ public function setAccountNumber($accountNumber)
+        $node = $document->createElement('BillThirdParty');
+
+        $btpNode = $node->appendChild($document->createElement('BillThirdPartyShipper'));
+        $btpNode->appendChild($document->createElement('AccountNumber', $this->getAccountNumber()));
```
Can we offload in here also to custom entities? So a 'BillThirdPartyShipper' with it's own `toNode` method? It feels more like following the existing flow than creating sub entities inside this class.

scottcwilson commented:
Argh I didn't realize there was an outstanding PR for this stuff.  I just submitted my own PR to do it.

thijsw commented:
@scottcwilson  Ahh too bad, and it's not the most fun work to do either... I must say I was a bit sloppy in not fixing the addressed issues in this PR otherwise I bet it'd be merged a lot sooner.

stefandoorn commented:
Hm, in the meanwhile we have 3 PR's actually on this subject. One by @thijsw, another one by @stephenjwinn and a recent one of @scottcwilson. Could you maybe team up in order to implement it? :-) I'm not using UPS actively right now and having time constraints, but can do reviewing and merging.

scottcwilson commented:
Please let me know if this PR should be closed @thijsw

stephenjwinn commented:
You can close mine

On Thu, Aug 16, 2018, 10:47 PM Scott C Wilson <notifications@github.com>
wrote:

> Please let me know if this PR should be closed @thijsw
> <https://github.com/thijsw>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gabrielbull/php-ups-api/pull/149#issuecomment-413742132>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEejDyaylRPE8P6C3W5yOOHvsFiA9nNaks5uRi68gaJpZM4MRH6F>
> .
>

scottcwilson commented:
Thank you @stephenjwinn.  Done.

gabrielbull commented:
@thijsw This library has no maintainer and is looking for one.
